axelor-human-resource/src/main/java/com/axelor/apps/hr/web/expense/ExpenseController.java

```diff
@@ -96,6 +97,21 @@ public class ExpenseController {
 
   private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+  public void autoCalculateTotalTaxFromProductAndTotalAmount(
+      ActionRequest request, ActionResponse response) {
+    ExpenseLine expenseLine = request.getContext().asType(ExpenseLine.class);
+    if (expenseLine != null) {
+      if (expenseLine.getExpenseProduct().getTaxComputeModeSelect() == ProductRepository.NO_TAX) {
```

Comment:
I think this can throw a NPE if expenseProduct is null, can you check ?

Also I think you can move this part to the service method, as it is a functional check.